app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesDao.java

```diff
@@ -52,27 +52,35 @@ public void save(RecentSearch recentSearch) {
 
     /**
      * This method is called on confirmation of delete recent searches.
-     * It deletes latest 10 recent searches from the database
-     * @param recentSearchesStringList list of recent searches to be deleted
+     * It deletes all recent searches from the database
      */
-    public void deleteAll(List<String> recentSearchesStringList) {
+    public void deleteAll() {
+        Cursor cursor = null;
         ContentProviderClient db = clientProvider.get();
-        for (String recentSearchName : recentSearchesStringList) {
-            try {
-                RecentSearch recentSearch = find(recentSearchName);
-                if (recentSearch.getContentUri() == null) {
-                    throw new RuntimeException("tried to delete item with no content URI");
-                } else {
-                    Timber.d("QUERY_NAME %s - delete tried", recentSearch.getContentUri());
-                    db.delete(recentSearch.getContentUri(), null, null);
-                    Timber.d("QUERY_NAME %s - query deleted", recentSearch.getQuery());
+        try {
+            cursor = db.query( RecentSearchesContentProvider.BASE_URI, Table.ALL_FIELDS,
```

Comment:
Horizontal whitespace - [GSG 4.6.2](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace)

```suggestion
            cursor = db.query(RecentSearchesContentProvider.BASE_URI, Table.ALL_FIELDS,
```

It might be an idea to put each argument of `db.query` on a different line so it looks like
```
cursor = db.query(
    RecentSearchesContentProvider.BASE_URI,
    Table.ALL_FIELDS,
    null,
    new String[]{},
    Table.COLUMN_LAST_USED + " DESC"
);
```